### PR TITLE
sharedWithDisplayName & sharedWithAvatar

### PR DESCRIPTION
**Pull request by ArtificialOwl**
Add displayName and Avatatar to IShare.

### CODE REVIEW

**Comment by daita** [comment]
selection_006

This PR fix https://github.com/nextcloud/circles/issues/78 and allow Circles to display its own avatar

**Review by LukasReschke — DISMISSED** [review]
Can I haz some unit tests on the new functions in Share.php? 😇

File: lib/public/Share/IShare.php [hunk 1]
```diff
@@ -168,6 +168,40 @@ public function setSharedWith($sharedWith);
 	public function getSharedWith();
 
 	/**
+	 * Set the display name of the receiver of this share.
+	 *
+	 * @param string $displayName
+	 * @return \OCP\Share\IShare The modified object
+	 * @since 9.0.0
```
Comment (nickvergessen) [inline lib/public/Share/IShare.php]: 13 everywhere

**Review by nickvergessen — DISMISSED** [review]

File: lib/public/Share/IShare.php (hunk 1 above)
Comment (daita) [inline lib/public/Share/IShare.php]: can I go 12.0.1 ?

Comment (nickvergessen) [inline lib/public/Share/IShare.php]: I don't think we want to backport this 🙈

Comment (daita) [inline lib/public/Share/IShare.php]: Why is wrong on having this on 12.0.1 ?

Comment (LukasReschke) [inline lib/public/Share/IShare.php]: We only backport serious bug fixes usually 😄

Comment (daita) [inline lib/public/Share/IShare.php]: ok

Comment (daita) [inline lib/public/Share/IShare.php]: @nickvergessen fixed

**Comment by daita** [comment]
you know have few tests @LukasReschke

**Comment by daita** [comment]
Tested with Circles, working !

We need to edit _https://apps.nextcloud.com/schema/apps/info.xsd_ for 
```
	<collaboration>
		<collaborators>
			<searchPlugins>
				<searchPlugin>
					<class></class>
					<shareType></shareType>
				</searchPlugin>
			</searchPlugins>
		</collaborators>
	</collaboration>
```

File: lib/public/Share/IShare.php [hunk 2]
```diff
@@ -167,6 +167,40 @@ public function setSharedWith($sharedWith);
...
+	 * Set the avatar of the receiver of this share.
+	 *
+	 * @param string $src
+	 * @return \OCP\Share\IShare The modified object
+	 * @since 13.0.0
+	 */
+	public function setSharedWithAvatar($src);
+
+	/**
+	 * Get the display name of the receiver of this share.
```
Comment (MorrisJobke) [inline lib/public/Share/IShare.php]: I guess you mean avatar.

**Comment by MorrisJobke** [comment]
@daita  Should we try to get this in 13 or is 14 fine?

**Comment by daita** [comment]
14 is fine

**Comment by MorrisJobke** [comment]
Rebased on current master, squashed some commits and updated the `@since` PHPDoc tags.

**Review by MorrisJobke — APPROVED** [review]
The avatars work nicely 👍

**Comment by MorrisJobke** [comment]
cc @nextcloud/sharing

**Comment by schiessle** [comment]
I just tried it... but no avatars. Maybe something got lost during the rebase?

**Comment by MorrisJobke** [comment]
@daita Also there is another merge conflict :(

**Comment by MorrisJobke** [comment]
Merge conflict caused by 2c073dc53d24c82dd4aaf315327d08463f2c7050 - maybe @danxuliu could help to rebase this.

**Comment by danxuliu** [comment]
>  Merge conflict caused by 2c073dc - maybe @danxuliu could help to rebase this.

I have rebased again and fixed the conflict; now the display name provided by the circle share is used and, if none is given, then the old code to extract it from `share_with` is used.

>  I just tried it... but no avatars. Maybe something got lost during the rebase?

I have fixed that too; avatars were not shown in the list due to https://github.com/nextcloud/server/pull/8843/commits/ed4b4458eb8bce2e461d952f11756e24c4e7f3c6, as since that commit `modSeed` was `false` when using circle avatars, and thus the `imageplaceholderseed` CSS class was not set.

Note, however, that the default avatar is still shown in the share dialog; showing the circle avatar there would require adding it [in the Circles app](https://github.com/nextcloud/circles/blob/748993ed70fcaa60ecf8d6e4c23d2c70e0b0d31e/lib/Collaboration/v1/CollaboratorSearchPlugin.php#L71) and then [using it from the dialog](https://github.com/nextcloud/server/blob/203bf51543f7d1866efeaa9decd2817cb41b3c3e/core/js/sharedialogview.js#L424-L429).

**Comment by danxuliu** [comment]
Oops, tests are failing :-) I will fix them.

**Comment by danxuliu** [comment]
>  Oops, tests are failing :-) I will fix them.

I forgot to push... :facepalm:

Anyway, I made a good use of my slip by adding more unit tests ;-) I have also rebased the commits onto current master.

This pull request should be ready for review again!

**Comment by MorrisJobke** [comment]
@rullzer @danxuliu Mind to give a final review? Or should we push it to 15?

**Comment by MorrisJobke** [comment]
> @rullzer @danxuliu Mind to give a final review? Or should we push it to 15?

Ping 🏓

**Review by danxuliu — APPROVED** [review]

**Pull request merged**